app/assets/javascripts/Components/Modals/filter_modal.jsx

```diff
@@ -0,0 +1,325 @@
+import {SingleSelectDropDown} from "../../DropDownMenu/SingleSelectDropDown";
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+
+export class FilterModal extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  onToggleOptionTas = user_name => {
+    const newArray = [...this.props.filterData.tas];
+    if (newArray.includes(user_name)) {
+      this.props.mutateFilterData({
+        ...this.props.filterData,
+        tas: newArray.filter(item => item !== user_name),
+      });
+    } else {
+      newArray.push(user_name);
+      this.props.mutateFilterData({
+        ...this.props.filterData,
+        tas: newArray,
+      });
+    }
+  };
+
+  onClearSelectionTAs = () => {
+    this.props.mutateFilterData({
+      ...this.props.filterData,
+      tas: [],
+    });
+  };
+
+  onClearSelectionTags = () => {
+    this.props.mutateFilterData({
+      ...this.props.filterData,
+      tags: [],
+    });
+  };
+
+  onToggleOptionTags = tag => {
+    const newArray = [...this.props.filterData.tags];
+    if (newArray.includes(tag)) {
+      this.props.mutateFilterData({
+        ...this.props.filterData,
+        tags: newArray.filter(item => item !== tag),
+      });
+    } else {
+      newArray.push(tag);
+      this.props.mutateFilterData({
+        ...this.props.filterData,
+        tags: newArray,
+      });
+    }
+  };
+
+  renderTasDropdown = () => {
+    if (this.props.role === "Instructor") {
+      let tas = this.props.tas.map(option => {
+        return {key: option[0], display: option[0] + " - " + option[1]};
+      });
+      return (
+        <div className={"filter"}>
+          <p>{I18n.t("activerecord.models.ta.other")}</p>
+          <MultiSelectDropdown
+            title={"Tas"}
+            options={tas}
+            selected={this.props.filterData.tas}
+            onToggleOption={this.onToggleOptionTas}
+            onClearSelection={this.onClearSelectionTAs}
+          />
+        </div>
+      );
+    }
+  };
+
+  renderTagsDropdown = () => {
+    let options = [];
+    if (this.props.available_tags.length !== 0) {
+      options = options.concat(
+        this.props.available_tags.map(item => {
+          return {key: item.name, display: item.name};
+        })
+      );
+    }
+    if (this.props.current_tags.length !== 0) {
+      options = options.concat(
+        this.props.current_tags.map(item => {
+          return {key: item.name, display: item.name};
+        })
+      );
+    }
+    return (
+      <MultiSelectDropdown
+        title={"Tags"}
+        options={options}
+        selected={this.props.filterData.tags}
+        onToggleOption={this.onToggleOptionTags}
+        onClearSelection={this.onClearSelectionTags}
+      />
+    );
+  };
+
+  rangeFilter = (min, max, title, onMinChange, onMaxChange) => {
+    return (
+      <div className={"filter"}>
+        <p>{title}</p>
+        <div className={"range"} data-testid={title}>
+          <input
```

Comment:
Add an [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) property here, which can be of the form `{title} - Min`, and one for the `Max` input below.

Also, the strings `Min` and `Max` should also be internationalized (see `placeholder` below) since they are user-facing.